### PR TITLE
PLAT-97898: Refactor sampler code for better reuse across UI libraries

### PR DESCRIPTION
**Pull request by JayCanuck**
Initial migration of common code from Storybook Sampler, including:
  * Storybook-specific babel/webpack configurations
  * Storybook addons (actions, knobs, info) pre-configured for Enact-based usage and simplified registration
  * Common utilities useful for samples (proptables, nullify, emptify, loadStories, etc.)

As a result, the Sampler architecture is now primarily composed of just a theme environment decorator and the stores themselves.

Enact-DCO-1.0-Signed-off-by: Jason Robitaille <jason.robitaille@lge.com>

### CODE REVIEW

File: addons/knobs/configure.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import {withKnobs as addonWithKnobs} from '@storybook/addon-knobs';
+
+function withKnobs(opts) {
+	return addonWithKnobs({
+		// debounce: {wait: 500}, // Same as lodash debounce.
```
Comment (germboy) [inline addons/knobs/configure.js]: I know this PR simply calls for copying the enact sampler/storybook config, but I'm wondering if it's a good time to clean up some lingering commented-out code. I'm not sure the purpose of this (or if it's needed for any reason). Should this get cleaned up?

File: addons/knobs/number.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import {number as numberKnob} from '@storybook/addon-knobs';
+
+// import nullify from '../utils/nullify.js';
```
Comment (germboy) [inline addons/knobs/number.js]: Same question about cleaning this up as above.

**Comment by JayCanuck** [comment]
Removed the unused commented-out code. As far as `package-lock.json`, I personally prefer to include them. The format is in a stable state and useful for viewing dependency trees. Have included them on `@enact/dev-utils`, `@enact/cli` and other packages without issue.

**Comment by germboy** [comment]
That works for me, thanks! 👍

File: addons/knobs/configure.js (hunk 1 above)
Comment (Djspaceg) [reply addons/knobs/configure.js]: I'd originally left it in case we wanted to restore it, but we haven't needed it yet, so it's fine to dispense with it.

**Comment by Djspaceg** [comment]
I like this consolidation. :)

**Pull request merged**